### ISSUE TITLE
RSEM generates zero byte size genomic bam file

### ISSUE THREAD

**Issue by arozhok** [alexdobin/STAR#750]
Running a typical rsem-calculate-expression command with the --output-genome-bam option. RNA-seq paired-end. STAR as aligner. Index generated from genome reference for STAR. Everything works, it generates counts and outputs two bam files. Trascriptomic bam looks ok. Genomic bam is a file of size zero bytes

**Comment by alexdobin** [comment]
Hi @arozhok 

is STAR called directly by RSEM? I am not sure what options are used. Do you see Log.out file from the STAR run - please send it to me.

Cheers
Alex

**Comment by arozhok** [comment]
Hi Alex,
Thanks for your reply. I use default RSEM options and the following command to launch:
rsem-calculate-expression -p $ncores --paired-end --append-names --output-genome-bam --star $file_left $file_right $rsem_index $prefix
the log.out file attached.
Thanks!
[R1Log.out.txt](https://github.com/alexdobin/STAR/files/3672263/R1Log.out.txt)

**Comment by arozhok** [comment]
Hi Alex,
Seems like I have some some way around the problem by specifying the option
_--star-output-genome-bam_ instead of _--output-genome-bam_. So now instead of the output file R1.genome.bam, which is zero bytes size, it outputs a file R1.STAR.genome.bam of a believable size. I am not yet sure if this new file is identical what the previous was supposed to be, though.

Andrew

**Comment by alexdobin** [comment]
Hi Andrew,

the parameters that RSEM uses for STAR call (from the R1Log.out file)
contain 
--outSAMtype BAM   Unsorted
which should output the R1.temp/R1Aligned.out.bam. Not sure why it's empty, maybe RSEM is doing something with it afterward - it gets renamed.

I usually run STAR standalone, and the RSEM just takes the Aligned.toTranscriptome.bam file.
This way we have full control of STAR parameters. The parameters that RSEM uses are:
outSAMunmapped                    Within   
outFilterType                              BySJout
outFilterMultimapNmax             20
outFilterMismatchNmax             999
outFilterMismatchNoverLmax    0.04
alignIntronMin                           20
alignIntronMax                          1000000
alignMatesGapMax                   1000000
alignSJoverhangMin                  8
alignSJDBoverhangMin              1
sjdbScore                                   1
quantMode                               TranscriptomeSAM   

Cheers
Alex

**Comment by arozhok** [comment]
Hi Alex,
I do see that file transiently when RSEM is working. But then it disappears and I get R1.genome.bam and R1.transcript.bam in the main working directory whereby R1.genome.bam is zero bytes. However, I found that the option I used _--star-output-genome-bam_ instead of _--output-genome-bam_ works for me nicely, as it produces a normal genome bam plus it sorts the bam files. So I guess I will stick with this strategy. Thanks a lot for assistance!
Andrew
